js/adslot.jsx

```diff
@@ -1,75 +1,80 @@
-    adUnit: React.PropTypes.string,
-    sizes: React.PropTypes.arrayOf(
-      React.PropTypes.oneOfType([
-        React.PropTypes.arrayOf(React.PropTypes.number),
-        React.PropTypes.string,
-      ])
+  static propTypes = {
+    dfpNetworkId: PropTypes.string,
+    adUnit: PropTypes.string,
+    sizes: PropTypes.arrayOf(
+      PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.number),
+        PropTypes.string,
+      ]),
     ),
-    renderOutOfThePage: React.PropTypes.bool,
-    sizeMapping: React.PropTypes.arrayOf(React.PropTypes.object),
-    fetchNow: React.PropTypes.bool,
-    targetingArguments: React.PropTypes.object,
-    onSlotRender: React.PropTypes.func,
-    shouldRefresh: React.PropTypes.func,
-    slotId: React.PropTypes.string,
-  },
-
-  contextTypes: {
-    dfpNetworkId: React.PropTypes.string,
-    dfpAdUnit: React.PropTypes.string,
-    dfpSizeMapping: React.PropTypes.arrayOf(React.PropTypes.object),
-    dfpTargetingArguments: React.PropTypes.object,
-  },
-
-  getDefaultProps() {
-    return {
-      fetchNow: false,
+    renderOutOfThePage: PropTypes.bool,
+    sizeMapping: PropTypes.arrayOf(PropTypes.object),
+    fetchNow: PropTypes.bool,
+    targetingArguments: PropTypes.object,
+    onSlotRender: PropTypes.func,
+    shouldRefresh: PropTypes.func,
+    slotId: PropTypes.string,
+    objectId: PropTypes.string,
+  };
+
+  static contextTypes = {
+    dfpNetworkId: PropTypes.string,
+    dfpAdUnit: PropTypes.string,
+    dfpSizeMapping: PropTypes.arrayOf(PropTypes.object),
+    dfpTargetingArguments: PropTypes.object,
+  };
+
+  static defaultProps = {
+    fetchNow: false,
+  };
+
+  constructor(props) {
+    super(props);
+    this.generateSlotId = this.generateSlotId.bind(this);
+    this.getSlotId = this.getSlotId.bind(this);
+    this.mapContextToAdSlotProps = this.mapContextToAdSlotProps.bind(this);
+    this.slotShouldRefresh = this.slotShouldRefresh.bind(this);
+    this.slotRenderEnded = this.slotRenderEnded.bind(this);
+    this.state = {
+      slotId: this.props.slotId || this.generateSlotId(),
     };
-  },
-
-  getInitialState() {
-    return { slotId: this.generateSlotId() };
-  },
+  }
 
   componentDidMount() {
     this.registerSlot();
-  },
+  }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.hasOwnProperty('objectId')) {
+    if (nextProps.objectId) {
       const state = this.state;
       state.slotId = this.generateSlotId();
       this.unregisterSlot();
       this.setState(state);
       this.registerSlot();
     }
-  },
+  }
+
   componentWillUnmount() {
     this.unregisterSlot();
-  },
+  }
 
   getSlotId() {
     return this.props.slotId || this.state.slotId;
-  },
+  }
 
   generateSlotId() {
-    let slotId = this.props.slotId;
-    if (slotId === undefined) {
-      const seconds = (Date.now && Date.now() || new Date().getTime()) / 1000;
-      slotId = `adSlot-${seconds}`;
-    }
+    const slotId = `adSlot-${dynamicAdCount}`;
```

Comment:
We can get it to two lines based on you ESLint config. Otherwise we get: `Unary operator '++' used. (no-plusplus)`. I push the change.